### PR TITLE
fix: rebuffering percentage inflated if client ads fail to load

### PR DESCRIPTION
**Pull request by daytime-em**

### CODE REVIEW

File: library/src/main/java/com/mux/stats/sdk/muxstats/MuxStatsSdkMedia3.kt [hunk 1]
```diff
@@ -101,6 +101,15 @@ class AdCollector private constructor(
    * Call when ad playback starts
    */
   fun onStartPlayingAds() {
+    // edge case: if an IMA mid/postroll ad fails to load in time, the ExoPlayer will go into the
+    //   BUFFERING state and we'll be in REBUFFERING, for one final attempt to get the ads.
+    //   If that fails, the player will go into an adbreak for a few millis and then continue with
+    //   content. The events all really happen and there really is a short playback interuption so
+    //   it's good to count it all
+    if (muxPlayerState == MuxPlayerState.REBUFFERING) {
```
Comment (andrewjl-mux) [inline library/src/main/java/com/mux/stats/sdk/muxstats/MuxStatsSdkMedia3.kt:109]: Would it make sense to add a test for handling this edge case? To avoid it potentially getting reverted later.

Comment (daytime-em) [reply library/src/main/java/com/mux/stats/sdk/muxstats/MuxStatsSdkMedia3.kt:109]: Yes, it totally would thanks for pointing that out

File: library/src/main/java/com/mux/stats/sdk/muxstats/MuxStatsSdkMedia3.kt [hunk 2]
```diff
@@ -101,6 +101,15 @@ class AdCollector private constructor(
...
+      stateCollector.pause()
```
Comment (cjpillsbury) [inline library/src/main/java/com/mux/stats/sdk/muxstats/MuxStatsSdkMedia3.kt:110]: question(non-blocking): any concerns with failing to unpause later (e.g. due to other jank conditions?) thereby deflating rebuffering?

File: library/src/test/java/com/mux/stats/sdk/muxstats/AdCollectorTests.kt [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+    // function under test
+    adCollector.onStartPlayingAds()
+
+    Assert.assertNotEquals(
+      "state should not be REBUFFERING after onStartPlayingAds",
+      MuxPlayerState.REBUFFERING, stateCollector.muxPlayerState
+    )
+
+    Assert.assertTrue(
```
Comment (cjpillsbury) [inline library/src/test/java/com/mux/stats/sdk/muxstats/AdCollectorTests.kt:42]: suggestion(non-blocking): As long as this has been smoke tested, I don't think this should be a blocker, but we should probably have 1+ test case(s) validating buffering after leaving ad playing state, ideally that adequately represent what can happen in real world usage, right?

**Review by cjpillsbury — APPROVED** [review]
Approving with a couple of nb callouts.

File: library/src/main/java/com/mux/stats/sdk/muxstats/MuxStatsSdkMedia3.kt (hunk 2 above)
Comment (daytime-em) [reply library/src/main/java/com/mux/stats/sdk/muxstats/MuxStatsSdkMedia3.kt:110]: Nope, it would be ok. We technically "unpause" when we get `adbreakstart` because `MuxPlayerState` goes from `PAUSED` to `AD_BREAK_STARTED` or something like that.

File: library/src/test/java/com/mux/stats/sdk/muxstats/AdCollectorTests.kt (hunk 3 above)
Comment (daytime-em) [reply library/src/test/java/com/mux/stats/sdk/muxstats/AdCollectorTests.kt:42]: I think we'd be ok, becuase that `pause()` call will end the rebuffering state at the beginning of the ad break (mentioned in other comment). And the way that it "ends the rebuffering state" is twofold: 1) setting our overall `muxPlayerState` thing to `PAUSED` 2) if `muxPlayerState` was `REBUFFERING` before that, it sends `rebufferend` to the core (and backend and dashboard)

File: library/src/main/java/com/mux/stats/sdk/muxstats/MuxStatsSdkMedia3.kt (hunk 2 above)
Comment (daytime-em) [reply library/src/main/java/com/mux/stats/sdk/muxstats/MuxStatsSdkMedia3.kt:110]: .. and the other part of your question: the rebuffer tracker doesn't check for `pause` so it's safe to send from that perspective.

A `pause` event is also not-wrong here, since content stopped playing. After the ad is over, `playing` will be sent, which will unpause us, and media3 makes it easy to protect that process from jank
